src/dpl/include/dpl/Objects.h

```diff
@@ -0,0 +1,266 @@
+/////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2024, Precision Innovations Inc.
+// All rights reserved.
+//
+// BSD 3-Clause License
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Coordinates.h"
+#include "Opendp.h"
+
+namespace dpl {
+
+using odb::dbOrientType;
+using odb::dbSite;
+using odb::Rect;
+
+class MasterEdge
+{
+ public:
+  MasterEdge(unsigned int type, const Rect& box);
+  unsigned int getEdgeType() const;
+  const Rect& getBBox() const;
+
+ private:
+  unsigned int edge_type_idx_{0};
+  Rect bbox_;
+};
+
+class Master
+{
+ public:
+  bool isMultiRow() const;
+  const std::vector<MasterEdge>& getEdges();
+  Rect getBBox();
+  void setMultiRow(bool in);
+  void addEdge(const MasterEdge& edge);
+  void clearEdges();
+  void setBBox(Rect box);
+
+ private:
+  Rect boundary_box_;
+  bool is_multi_row_ = false;
+  std::vector<MasterEdge> edges_;
+};
+
+class Pin;
+class Group;
+
+class Node
+{
+ public:
+  enum Type
+  {
+    UNKNOWN,
+    CELL,
+    TERMINAL,
+    MACROCELL,
+    FILLER
```

Comment:
I suppose this is needed by dpo but it is strange to have these lumped into a single class as most of these are not moveable.  Something for another PR

Reply:
Agreed.